obspy/core/util/base.py

```diff
@@ -60,6 +59,30 @@
 NATIVE_BYTEORDER = _sys_is_le and '<' or '>'
 
 
+# apply simple memoization cache on load_entry_points
+# this function cannot go in decorator.py due to circular import issues
+def _load_entry_point_decorator(func):
+    """
+    Decorate pkg_resources' load_entry_point function to cache outputs.
+    """
+    cache = {}
+    expected_keys = ('dist', 'group', 'name')
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # make a str key from the inputs to func for hashing in cache
+        input_dict = inspect.getcallargs(func, *args, **kwargs)
+        hash_str = '/'.join([input_dict[x] for x in expected_keys])
+        if hash_str not in cache:
+            cache[hash_str] = func(*args, **kwargs)
+        return cache[hash_str]
+
+    return wrapper
+
+
+load_entry_point = _load_entry_point_decorator(load_entry_point)
```

Comment:
> yea then this is not used by ObsPy

It actually is, that's what puzzles me.. (In the example I've added a print statement inside the decorator)

```python
In [1]: cd git/obspy/obspy/io/mseed/tests/data/
/home/megies/git/obspy/obspy/io/mseed/tests/data

In [2]: st = read('two_channels.mseed')
in _load_entry_point_decorator
in _load_entry_point_decorator

In [3]: st = read('two_channels.mseed')
in _load_entry_point_decorator
in _load_entry_point_decorator

In [4]: st = read('two_channels.mseed')
in _load_entry_point_decorator
in _load_entry_point_decorator

In [5]: from pkg_resources import load_entry_point

In [6]: load_entry_point('obspy', 'obspy.plugin.waveform.MSEED', 'readFormat')
Out[6]: <function obspy.io.mseed.core._read_mseed>

In [7]: from obspy.core.util.base import load_entry_point

In [8]: load_entry_point('obspy', 'obspy.plugin.waveform.MSEED', 'readFormat')
in _load_entry_point_decorator
Out[8]: <function obspy.io.mseed.core._read_mseed>

In [9]: load_entry_point('obspy', 'obspy.plugin.waveform.MSEED', 'readFormat')
in _load_entry_point_decorator
Out[9]: <function obspy.io.mseed.core._read_mseed>

In [10]: from pkg_resources import load_entry_point

In [11]: load_entry_point('obspy', 'obspy.plugin.waveform.MSEED', 'readFormat')
Out[11]: <function obspy.io.mseed.core._read_mseed>
```